Prepare repo

hansvancalster opened the pull request:
This PR will do some restructuring of folders and update README.md

florisvdh commented:
@hansvancalster I'll look at it today

florisvdh commented on README.md:
```diff
@@ -92,71 +194,35 @@ create_sfp(title = "titel van het protocol",
-You will now have a local clone of the remote repository as an RStudio project. The file `protocols.Rproj` will be automatically created as well as a `.git` directory (do not make changes in this directory - it is where version control takes place). 
-The same directories and files which can be seen on the remote (https://github.com/inbo/protocols) will be copied to your local drive. 
-Whenever you want to work in the project, you need to double-click `protocols.Rproj` file to start the RStudio project. 
+Some useful resources for self-learning of **Rmarkdown**:
```
Other suggestions: https://bookdown.org/yihui/rmarkdown/ and https://bookdown.org/yihui/bookdown/ . Rather as a reference.

hansvancalster replied:
See 078cb8a

florisvdh commented on README.md:
```diff
@@ -92,71 +194,35 @@ create_sfp(title = "titel van het protocol",
+Some useful resources for self-learning of **Rmarkdown**:
 
-To refer to files use the `rprojroot` package. Here is an example to refer to the utils folder:
+[tutorial rmarkdown](https://ourcodingclub.github.io/2016/11/24/rmarkdown-1.html)
```
best use bullets, this line looks a bit orphaned

hansvancalster commented on README.md:
```diff
@@ -2,51 +2,153 @@
   |- README.md <- up to date description of the repo content
-  |- docs <- contains rendered versions of protocols
+  |- docs <- can be used to locally inspect rendered versions of protocols, gitignored
   |- from_docx <- contains Microsoft Word docx protocols which were converted to 
                   markdown
```
I prefer to keep it as is. The docx files should not be git-ignored for archival reasons, availability of the file to both repo-admin and subject-matter specialists, and for checking if something went wrong with conversion.

The `docs` folder on the other hand, only serves for locally checking rendered result.

florisvdh replied:
> The docx files should not be git-ignored for archival reasons

I understand. Just an idea: when such a protocol has evolved to a newer version, IMO it would be good (and more consistent) to delete the older docx file. It will still be archived in the git repo (older commits). This would assist in providing just one version of each protocol in each release.

hansvancalster replied:
e82afc6

hansvancalster commented on README.md:
```diff
@@ -2,51 +2,153 @@
   |    |   |- 5_species
   |    |- project <- project-specific protocols
-  |    |- utils
+  |    |- management <- for repo admins only
```
see d666651

hansvancalster commented on README.md:
````diff
@@ -2,51 +2,153 @@
+  "src",
+  "project"
+  )
+```
````
eb08259

hansvancalster commented on README.md:
```diff
@@ -2,51 +2,153 @@
+Whenever a pull request is reviewed and finalized. 
+A repo-admin will merge the branch to the master and add general and specific tags. 
+The general tag is of the form `protocols-YYYY.number`.
+The specific tag is of the form `protocol-code-YYYY.number`.
```
906094f

hansvancalster commented on README.md:
```diff
@@ -2,51 +2,153 @@
+    - stage the files generated from the template in the git pane
+    - press commit button and add a commit message
+    - press the commit button
+    - press the push button
```
93e2b4e

hansvancalster commented on README.md:
```diff
@@ -2,51 +2,153 @@
+
+The workflow is as follows for a _new_ protocol:
+
+- make sure your local clone of the remote repository is up to date:
```
93e2b4e

hansvancalster commented on README.md:
```diff
@@ -2,51 +2,153 @@
+The master branch is protected and can only receive commits from reviewed pull requests.
+Development of a protocol (see [Workflow](#workflow)) is done in a protocol-specific branch that branches off the master branch.
+
+Protocols can depend on other protocols (protocol dependencies) only if these dependent protocols have been approved (and are published). 
```
See eef8bfa. Needs review...

hansvancalster commented on README.md:
````diff
@@ -2,51 +2,153 @@
-* chunk options: https://yihui.name/knitr/options/
-* chunk names: are optional but recommended especially if an output is generated by the chunk, do not use spaces or _ in chunk names (e.g. ```r name-of-chunk)
-* use 1 chunk for 1 (ggplot) figure or 1 table or other type of output that is printed 
+![](protocols-gitflow-model.png)
````
They look OK to me?

florisvdh replied:
The legend says 'start from template'. However, this is not the case for existing protocols, cf. the lower yellow circle.

hansvancalster commented:
Thanks for your review @florisvdh! I addressed most of your comments and they have certainly improved the README.md. I also followed your suggestion to add a CONTRIBUTING.md file and moved relevant parts from the README.md to that file. 

Can you have another look and approve if OK?

hansvancalster commented on README.md:
```diff
@@ -2,161 +2,132 @@
+Protocols can be either written in one or multiple languages. The functions in the `protocolhelper` package support Dutch (language = 'nl') and English (language = 'en').
 
-#### Using RStudio and git version control
+Together with a version number, a language identifier and the protocol-code, a unique version of a protocol will be identified.
```
It is not strictly necessary, but maybe the easiest thing to do. So I will change the text a bit. 

(Protocol-code + version number will be sufficient to uniquely identify a protocol if we assume that two language versions of the same version of a protocol will never be released at the same time (same merge commit). I think this is safe to assume.)

florisvdh replied:
> will be sufficient to uniquely identify a protocol if we assume that [...] will never be released at the same time (same merge commit)

Do you mean that the version number will be different between 'nl' and 'en', hence there is a distinction? I think we will have to decide whether we accept apparently evolved versions `sfp-401-2020.6` and `sfp-401-2020.7` which actually are each other's translations.

On the other hand, we won't have that so much. Perhaps an intermediate solution could be to add the language identifier when it is different from 'nl'.

hansvancalster replied:
Good point!
Maybe the following could work?
````
protocols-2020.6   sfp-401-nl-2020.6 #a first nl version
protocols-2020.7   sfp-401-nl-2020.7 #a second nl version
protocols-2020.8   sfp-401-en-2020.6 #translation of nl to en of version 2020.6
protocols-2020.9   sfp-401-nl-2020.9 #a third nl version (bump to 2020.9 instead of 2020.8)
`````

florisvdh replied:
I understand the idea, however I'm afraid it would create confusion about the meaning of a version number. Also, it would complicate management (= more error-prone). I think there would be no problem with:

`protocols-2020.8   sfp-401-en-2020.8 #translation of nl to en of version 2020.6`

It is more consistent with the expectation that version numbers are identical between tags that point to the same commit. To get a sensible timeline of one protocol, one would then need to do it _language-specific_, but I think that makes sense. I.e. filtering by either `sfp-401-nl` or `sfp-401-en` to see what the consecutive versions are.

Considering the gaps in version numbers this creates, seen from a language-specific protocol, note that also ongoing releases of other protocol-codes, e.g. `sfp-101`, will cause gaps in the version numbers of `sfp-401` anyway.

I think there is another opportunity of keeping version numbers and language identifiers independent: it effectively allows `sfp-401-nl` and `sfp-401-en` to be (somewhat) different in their contents as well, perhaps because of collaboration with other researchers, etc. This would make clear that both are actually different protocols, though still about the same subject and method.

hansvancalster replied:
OK, I think we go with your idea. In that case, somewhere in the translated protocol it should be mentioned from which protocol it is translated from (i.e. a reference to the protocol specific tag).

hansvancalster replied:
@ThierryO I'd like to hear your opinion on this too.

ThierryO replied:
I'd upgrade all language versions in the same PR. Hence they get the same version number. If you plan a translated version, use it to first update the current version. If you update the original version, then you should update the translations accordingly.

hansvancalster replied:
That's probably the neatest way to do it. @florisvdh do you agree?
That means we can drop the language identifier from the specific tag.

hansvancalster replied:
I've implemented this in 71fbb07

florisvdh replied:
Sure, we can go with this approach. It does narrow workflow flexibility and may request more effort from the contributor (enforcing parallel translation). On the other hand it makes the release cycle and protocol version numbering easier for the protocol user. Also, if content differs more than pure translation, this approach will imply a different protocol code, which makes that aspect more transparent.

Then, how will we cope with not-willingness to proceed with translating into a certain language, if an older version exists in that language? A reason may be that that translation is not longer needed at a certain point. For consistency, I suggest deleting it in the next development cycle; older versions remain on the website anyway. You may want to add such a step in the workflow, now that we've thought about it :wink:

hansvancalster replied:
Will at that to the workflow and after that is done I will merge this branch to master. Thanks to both of you for your help!

hansvancalster commented on README.md:
```diff
@@ -2,161 +2,132 @@
+For this specific purpose, we can make use of Rmarkdown's ability to recycle  specific parts of another protocol using a child chunk in the (parent) project-specific protocol.
 
-### Starting a new protocol
+Such child-reference will need to be version-specific if it is not to cause clashes with version numbering: a project-protocol at some version should not have evolving compiled counterparts.
```
Indeed, I didn't have a solution in mind when I wrote it. But I leave the technical stuff for later. If it appears not doable, we will have to rethink this.

hansvancalster commented on README.md:
```diff
@@ -2,161 +2,132 @@
-## Starting a new protocol
+We use a simple branching model. 
+The master branch is protected and can only receive commits from reviewed pull requests.
+Development of a protocol (see [Workflow](#workflow)) is done in a protocol-specific branch that branches off the master branch.
```
71fbb07

florisvdh commented on CONTRIBUTING.md:
```diff
@@ -188,8 +188,9 @@ Here is an example to refer to the project folder:
 
-Using online editing functionalities
+If you are not familiar with git and RStudio, there are still ways to contribute.
+The GitHub website has [online editing functionalities](https://help.github.com/en/github/managing-files-in-a-repository/editing-files-in-your-repository) which can be used to suggest changes to draft protocols. 
```
:+1:

florisvdh commented on CONTRIBUTING.md:
```diff
@@ -144,14 +144,16 @@ create_sfp(title = "titel van het protocol",
 If you plan a translated version, use it to first update the current version. 
 If you update the original version, then you should update the translations accordingly.
 
+In case one of the languages of a multi-language protocol is discontinued for some reason, the Rmarkdown files for the discontinued language should be deleted in the protocols repo (they are still available in the older commits of the protocols repo and the rendered version of all versions of protocols are kept at the `protocols-website` repo.
```
Fix typo: a closing bracket is to be added at the end here.
